### DISCUSSION TITLE
有几个关于响应器的问题想问一下，麻烦大佬解答一下

### DISCUSSION THREAD

**Discussion by meeger** [alibaba/formily#2437] [Q&A 2.x] [answered]
1.响应器中动作响应获取不到$deps的值，属性响应可以，动作响应中就是获取不到吗  
2.在动作响应中，如果想设置非当前字段的一些属性，是否只能通过下面这种方法$form.setFieldState("bbb", (field) => {
    field.setTitle("mmmm")
  })
3.是否能在动作响应器里使用form effect hooks和field effect hooks  以及一些自定义hook，该怎么使用  
4.表单设计器里表单的属性配置能否像字段的属性配置一样使用响应器

**Comment by janryWang** [comment, marked as answer]
1. 动作响应也可以的
2. setFieldState或者用$form.query查找字段，然后设置字段属性
3. 不支持生命周期钩子，其实被动联动都能解决的
4. 不能，响应器只支持字段级别的

**Reply by meeger** [reply]
大佬，再问一下
1.如果我想暴露一些表单相关的钩子该怎么处理呢，比如在formOnload或者formOnunload的时候允许用户自定义一些操作，是在某个地方收集用户的自定义操作，然后在creatForm的时候自己处理吗，还有更好的建议吗？

**Reply by meeger** [reply]
还有被动联动是怎么做的，我看表单的被动联动是使用了onFieldReact,但是在响应器里没法使用这个钩子该怎么办呢

**Reply by janryWang** [reply]
自己想办法实现吧，这种都是具体的实现逻辑而已，无非就是一些数据转换
